wls-gui-wahllokalsystem/tests/utils/stimmabgabevermerke/StimmabgabevermerkeTestDataFactory.ts

```diff
@@ -0,0 +1,72 @@
+import type { Stimmabgabevermerke } from "@/types/stimmabgabevermerke/Stimmabgabevermerke.ts";
+import type { Wahldaten } from "@/types/stimmabgabevermerke/Wahldaten.ts";
+import type { Builder } from "@tests/utils/Builder.ts";
+
+import { proxyBuilder } from "@tests/utils/Builder.ts";
+import { useCommonTestDataFactory } from "@tests/utils/common/CommonTestDataFactory.ts";
+
+import { EingenommenerWahlscheinStimmzettelartEnum } from "@/types/stimmabgabevermerke/EingenommenerWahlscheinStimmzettelartEnum.ts";
+
+const { generateRandomString, generateRandomNumber } =
+  useCommonTestDataFactory();
+
+export function useStimmabgabevermerkeTestDataFactory() {
+  function createWahldaten(): Wahldaten {
+    return {
+      eingenommeneWahlscheine: new Map([
+        [
+          EingenommenerWahlscheinStimmzettelartEnum.Klein,
+          generateRandomNumber(4),
+        ],
+      ]),
+      waehlerverzeichnisNummer: generateRandomNumber(1),
+      wahlID: generateRandomString(10),
+      wahlbezirkID: generateRandomString(10),
+    };
+  }
+
+  function createStimmabgabevermerke(): Stimmabgabevermerke {
+    const wahldatenOne = prepareWahldaten()
+      .eingenommeneWahlscheine(
+        new Map([[EingenommenerWahlscheinStimmzettelartEnum.Klein, 50]])
+      )
+      .build();
+    const wahldatenTwo = prepareWahldaten()
+      .eingenommeneWahlscheine(
+        new Map([[EingenommenerWahlscheinStimmzettelartEnum.Klein, 20]])
+      )
+      .build();
+    const wahldatenThree = prepareWahldaten()
+      .eingenommeneWahlscheine(
+        new Map([[EingenommenerWahlscheinStimmzettelartEnum.Klein, 30]])
+      )
+      .build();
```

Comment:
Und muss es wirklich klein sein? Wenn ja sollte dies entweder durch einen optionalen Parameter mit Default für Klein oder den Funktionsnamen sichtbar werden. In der Regeln hatte wir bei create viel Zufall drin.

```suggestion
    const wahldatenOne = prepareWahldaten()
      .eingenommeneWahlscheine(
        new Map([[EingenommenerWahlscheinStimmzettelartEnum.Klein, generateRandomNumber(2)]])
      )
      .build();
    const wahldatenTwo = prepareWahldaten()
      .eingenommeneWahlscheine(
        new Map([[EingenommenerWahlscheinStimmzettelartEnum.Klein, generateRandomNumber(2)]])
      )
      .build();
    const wahldatenThree = prepareWahldaten()
      .eingenommeneWahlscheine(
        new Map([[EingenommenerWahlscheinStimmzettelartEnum.Klein, generateRandomNumber(2)]])
      )
      .build();
```

Reply:
Ich hab die Test Funktion angepasst. Werte werden jetzt zufällige generiert, auch die enum werte. Wenn ein genauer enum wert gesetzt wird, macht das jetzt die test Funktion und nicht die factory